pkg/metrics/cluster.go

```diff
@@ -149,6 +186,40 @@ func CleanupMetricsForCluster(clusterName string) {
 	clusterSyncStatusDuration.DeleteLabelValues(clusterName)
 }
 
+// RecordEvictionQueueMetrics record the depth Of the EvictionQueue
+func RecordEvictionQueueMetrics(name string, depth float64) {
+	evictionQueueMetrics.WithLabelValues(name).Set(depth)
+}
+
+// RecordEvictionKindMetrics records eviction queue items by resource type
+// Increase count when true and decrease count when false
+func RecordEvictionKindMetrics(clusterName, resourceKind string, increase bool) {
+	if clusterName == "" || resourceKind == "" {
+		return
+	}
+
+	if increase {
+		evictionKindTotalMetrics.WithLabelValues(clusterName, resourceKind).Inc()
+	} else {
+		evictionKindTotalMetrics.WithLabelValues(clusterName, resourceKind).Dec()
+	}
+}
+
+// RecordEvictionProcessingMetrics records the processing delay and results of the eviction task
+func RecordEvictionProcessingMetrics(name string, err error, startTime time.Time) {
+	latency := utilmetrics.DurationInSeconds(startTime)
+	evictionProcessingLatency.WithLabelValues(name).Observe(latency)
+
+	result := utilmetrics.GetResultByError(err)
+	evictionProcessingTotal.WithLabelValues(name, result).Inc()
+}
+
+// RecordClusterHealthMetrics record cluster health indicators
+func RecordClusterHealthMetrics(unhealthyClusters int, failureRate float64) {
```

Comment:
This function is not being called; can it be deleted? Can the corresponding metrics also be deleted first?